apps/asap-server/src/controllers/users.ts

```diff
@@ -153,29 +153,44 @@ export default class Users {
     take: number;
     skip: number;
     search?: string;
-    filter?: string | string[];
+    filter?: string[];
   }): Promise<ListUserResponse> {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const { search, filter, ...opts } = options;
-
-    const { total, items: users } = await this.users.fetch({
-      ...opts,
-      ...(search
-        ? {
-            filter: {
-              or: search
-                .split(' ')
-                .filter(Boolean)
-                .flatMap((word) => [
-                  { path: 'data.displayName.iv', op: 'contains', value: word },
-                  { path: 'data.firstName.iv', op: 'contains', value: word },
-                  { path: 'data.lastName.iv', op: 'contains', value: word },
-                ]),
-            },
-          }
-        : {}),
-      sort: [{ path: 'data.displayName.iv' }],
-    });
+    const { take, skip, search, filter } = options;
+
+    const searchQ = (search || '')
+      .split(' ')
+      .filter(Boolean) // removes whitespaces
+      .reduce(
+        (acc: string, word: string) =>
+          acc.concat(
+            `contains(data/displayName/iv, '${word}') or contains(data/firstName/iv, '${word}') or `,
+          ),
+        '',
+      )
+      .slice(0, -4); // remove last ' or '
```

Comment:
We could concat to an array and use `.join(' or ')`.

Reply:
Hell yeah! Sound even better! 😏